_cite/plugins/google-scholar.py

```diff
@@ -42,13 +42,13 @@ def query(_id):
     for work in response:
         # create source
         source = {
-            "id": work.get("citation_id", ""),
+            "id": get_safe(work, "citation_id", ""),
             # api does not provide Manubot-citeable id, so keep citation details
-            "title": work.get("title", ""),
-            "authors": list(map(str.strip, work.get("authors", "").split(","))),
-            "publisher": work.get("publication", ""),
-            "date": work.get("year", "") + "-01-01",
-            "link": work.get("link", ""),
+            "title": get_safe(work, "title", ""),
+            "authors": list(map(str.strip, get_safe(work, "authors", "").split(","))),
+            "publisher": get_safe(work, "publication", ""),
+            "date": get_safe(work, "year", "") + "-01-01",
```

Comment:
That's a good question; if "work.year" for some reason wasn't present, should the value be "-01-01", "", `None`, or should there just be no `date` key at all?

Reply:
Keeping consistency with the Manubot date formatting, it should be `""`. Any nicer syntax to do that than this:

`"date": get_safe(work, "year", "") + "-01-01" if get_safe(work, "year", "") else "",`

Reply:
That's how I'd do it, but it is too bad that you have to evaluate `get_safe()` twice. There's a recent addition to Python (I believe it was introduced in 3.8), "assignment expressions", that get around this: https://peps.python.org/pep-0572/.

In this case you'd use it like:
```python
"date": f"{year_val}-01-01" if (year_val := get_safe(work, "year", "")) else ""
```

It is pretty new so it'd probably be unfamiliar to most Python developers, and it limits you to Python 3.8+, so it might not be a great idea to introduce it here.

Reply:
I ended up just doing this https://github.com/greenelab/lab-website-template/pull/198/commits/00fb8a9407f0dbbbe03d00fb310b6e8e39065fa7